language-extensions/dotnet-core-CSharp/src/managed/CSharpOutputDataSet.cs

```diff
@@ -51,15 +56,27 @@ public unsafe void ExtractColumns(DataFrame CSharpDataFrame)
             {
                 DataFrameColumn column = CSharpDataFrame.Columns[columnNumber];
 
+                // Determine the SQL data type for this column
+                // Prefer the incoming input column metadata when available so nvarchar metadata is preserved.
+                // Default to ANSI mapping to keep existing behavior when no metadata is supplied.
+                //
+                SqlDataType dataType = DataTypeMap[column.DataType];
+                ulong columnSize = (ulong)DataTypeSize[dataType];
+                if(inputColumns != null && inputColumns.ContainsKey(columnNumber))
```

Comment:
TryGetValue is a more efficient way of retrieving the metadata for column number x. However, I still dont understand how can data type of columns in the InputDataSet affect those in the OutputDataSet and that too indexed at same numbers?